astropy/utils/introspection.py

```diff
@@ -111,6 +121,19 @@ def minversion(module, version, inclusive=True, version_path='__version__'):
     True
     """
 
+    if isinstance(module, types.ModuleType):
+        module_name = module.__name__
+    elif isinstance(module, six.string_types):
+        module_name = module
+        try:
+            module = resolve_name(module_name)
+        except ImportError:
+            return False
```

Comment:
In other words, this should not be a replacement for

```
try:
    import scipy
    HAS_SCIPY = True
except ImportError:
    HAS_SCIPY = False
```

it's explicitly limited to just this case of checking for a specific minimum version of a package independent of whether *any* version of the package is available.